### PR TITLE
Feedback, about, settings screen, header hoc

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This adds:

Settings screen - on native to match designs
Feedback screen - that replaces log send with text feedback that will be put in the status tab.
About screen - Shows the version of the app
Header hoc - Adds the title at the top and a back button. works with recompose stuff too.

### CODE REVIEW

**Comment by MarcoPolo** [comment]
@gabriel Think you can review the changes to the go code?

**Comment by keybase-ci-visdiff** [comment]
The commits 7666485c927a...8e66e21ca400 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490310681591-7666485c927a-8e66e21ca400-linux/Icon-Icon_Image-7666485c927a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490310681591-7666485c927a-8e66e21ca400-linux/Icon-Icon_Image-8e66e21ca400.png)  
   Icon-Icon_Image comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 7666485c927a...8e66e21ca400 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490312781475-7666485c927a-8e66e21ca400-linux/Icon-Icon_Image-7666485c927a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490312781475-7666485c927a-8e66e21ca400-linux/Icon-Icon_Image-8e66e21ca400.png)  
   Icon-Icon_Image comparison
</summary>

File: shared/settings/feedback-container.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    onSendFeedbackContained: ({sendLogs, feedback, onSendFeedback}) => () => onSendFeedback(feedback, sendLogs),
+  })
+)(Feedback)
+
+type State = {
+  sentFeedback: boolean,
+  feedback: ?string,
+}
+
+class FeedbackContainer extends Component<void, {}, State> {
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: I find this form more compact if your constructing just the state to fixed non-prop values

```
class FeedbackContainer extends Component<void, {}, State> {
state: {
   sentFeedback: false,
  feedback: null,
}

// no constructor
```

File: shared/settings/feedback-container.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  _onChangeFeedback = (feedback) => {
+    this.setState({feedback})
+  }
+
+  render () {
+    const onSendFeedback = (feedback, sendLogs) => {
+      logSend(feedback, sendLogs).then(logSendId => {
+        console.warn('logSendId is', logSendId)
+        this.setState({
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: I think this is a dangerous pattern. you have to check to ensure you're not dismounted else setState will complain. like click send feedback and back quickly

File: shared/settings/feedback-container.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        })
+      })
+    }
+
+    return <FeedbackWrapped showSuccessBanner={this.state.sentFeedback} onSendFeedback={onSendFeedback} onChangeFeedback={this._onChangeFeedback} feedback={this.state.feedback} />
+  }
+}
+
+export default connect(
+  () => ({}),
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: use null here too

File: shared/settings/feedback-container.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  (dispatch: Dispatch, {navigateUp}) => ({
+    title: 'Feedback',
+    onBack: () => dispatch(navigateUp()),
+  })
+)(HeaderHoc(FeedbackContainer))
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: might as well use compose

File: shared/settings/nav/index.native.js [hunk 5]
```diff
@@ -1,8 +1,84 @@
...
+        <Text type={'BodySmallSemibold'} style={itemTextStyle}>{text.toUpperCase()}</Text>
+        {!!badgeNumber && badgeNumber > 0 && <Badge badgeStyle={badgeStyle} badgeNumber={badgeNumber} />}
+      </Box>
+    </ClickableBox>
+  )
+}
+
+function SettingsNav ({badgeNumbers, selectedTab, onTabChange}: Props) {
+  return (
+    <Box style={styleNavBox}>
```
Comment (chrisnojima) [inline shared/settings/nav/index.native.js]: don't have to do this now but we could just use a [].map()

File: shared/settings/feedback-container.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/settings/feedback-container.js]: ah right, suggestions?

**Comment by keybase-ci-visdiff** [comment]
The commits 7666485c927a...c3268869b0f8 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490380205346-7666485c927a-c3268869b0f8-linux/Icon-Icon_Image-7666485c927a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490380205346-7666485c927a-c3268869b0f8-linux/Icon-Icon_Image-c3268869b0f8.png)  
   Icon-Icon_Image comparison
</summary>

File: shared/settings/feedback-container.js (hunk 2 above)
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: i typically call this._onLogsendDone, which is either null or set using componentDidMount/unmount

Comment (chrisnojima) [inline shared/settings/feedback-container.js]: or just set a flag isMounted and check that

File: shared/settings/feedback-container.js [hunk 6]
```diff
@@ -22,39 +22,45 @@ type State = {
...
+  state = {
+    sentFeedback: false,
+    feedback: null,
   }
 
   _onChangeFeedback = (feedback) => {
     this.setState({feedback})
   }
 
+  componentWillUnmount () {
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: you want this.mounted to be set in didMount, not initialized to it either

Comment (chrisnojima) [inline shared/settings/feedback-container.js]: then 👍

**Comment by keybase-ci-visdiff** [comment]
The commits 7666485c927a...8fe11b7fc70b introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490384201902-7666485c927a-8fe11b7fc70b-linux/Icon-Icon_Image-7666485c927a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490384201902-7666485c927a-8fe11b7fc70b-linux/Icon-Icon_Image-8fe11b7fc70b.png)  
   Icon-Icon_Image comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits cb51fbbd9b45...a190114c9d22 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490384599925-cb51fbbd9b45-a190114c9d22-linux/Icon-Icon_Image-cb51fbbd9b45.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490384599925-cb51fbbd9b45-a190114c9d22-linux/Icon-Icon_Image-a190114c9d22.png)  
   Icon-Icon_Image comparison
</summary>

**Pull request merged**